src/inference/src/dev/core_impl.cpp

```diff
@@ -455,64 +684,85 @@ std::vector<std::string> ov::CoreImpl::get_available_devices() const {
     return devices;
 }
 
-ov::RemoteContext ov::CoreImpl::create_context(const std::string& device_name, const AnyMap& args) const {
-    auto parsed = ov::parseDeviceNameIntoConfig(device_name, args);
+ov::RemoteContext ov::CoreImpl::create_context(const std::string& device_name, const AnyMap& params) const {
+    auto parsed = ov::parseDeviceNameIntoConfig(device_name, params);
     return get_plugin(parsed._deviceName).create_context(parsed._config);
 }
 
-ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& device_name, const ov::AnyMap& config) const {
-    std::vector<std::string> supportedConfigKeys;
+ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& full_device_name,
+                                                const ov::AnyMap& user_properties) const {
+    static const std::vector<std::string> core_level_properties = {
+        ov::cache_dir.name(),
+        ov::force_tbb_terminate.name(),
+        // auto-batch properties are also treated as core-level
+        ov::auto_batch_timeout.name(),
+        ov::hint::allow_auto_batching.name(),
+    };
+
+    const auto flattened = ov::parseDeviceNameIntoConfig(full_device_name, user_properties);
+    const std::string& device_name = flattened._deviceName;
+    const auto& flattened_config = flattened._config;
+    ov::AnyMap supported_config, options;
+
+    // fill 'options' to provide more information to ICore::get_property calls
+    {
+        auto priority_prop_name = get_device_priority_property(device_name).prop_name;
+        auto it = flattened_config.find(priority_prop_name);
+        if (it != flattened_config.end())
+            options[it->first] = it->second;
+        else if (device_name == "HETERO") {
+            // TODO: remove together with API 1.0
+            priority_prop_name = "TARGET_FALLBACK";
+            it = flattened_config.find(priority_prop_name);
+            if (it != flattened_config.end())
+                options[it->first] = it->second;
+        } else if (device_name == "BATCH") {
+            // TODO: remove together with API 1.0
+            priority_prop_name = CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG);
+            it = flattened_config.find(priority_prop_name);
+            if (it != flattened_config.end())
+                options[it->first] = it->second;
+        }
+    }
+
+    // virtual plugins should bypass core-level properties to HW plugins
+    // so, we need to report them as supported
+    std::vector<std::string> supported_config_keys = core_level_properties;
```

Comment:
Core_level_property will be included in device's supported properties? But some of them will never be applied into any device, e.g. ov::force_tbb_terminate.name().

Reply:
'get_supported_properties' is called from meta plugins. So, the chain of calls is the following:
- AUTO / HETERO / etc calls 'get_supported_properties' for each `device` to understand what properties need to be passed down to `get_core()->compile_model(model, device, properties)`
- `get_supported_properties` returns properties for `Core` itself and for `device`
- during `get_core()->compile_model(model, device, properties)`:
    -  `get_core()` will filter out core-level properties
    - pass to actual `get_plugin().compile_model(model, device, properties_without_core_level_properties)` only device-supported properties.